pkg/download/downloader_test.go

```diff
@@ -58,23 +59,20 @@ func Test_extractZIP(t *testing.T) {
 	for _, tt := range tests {
 		// Zip has just one file named 'foo'
 		zipSrc := filepath.Join(testdataPath(), tt.in)
-		zipDst, err := ioutil.TempDir("", "")
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer os.RemoveAll(zipDst)
+		tmpDir, cleanup := testutil.NewTempDir(t)
```

Comment:
This is a weird ask but when we (at Google) do syntactic improvements like these, we also try to make sure future people don't go using the old way when there's a new version available. We usually add checks in the code for such things.

So I wonder if there's any point in making a hack/ script like `validate-tests-dont-use-ioutil-tempdir.sh` that does something like:

```sh
out="$(grep -ERI 'ioutil\.TempDir'  . | grep -v tempdir_test.go )"
if [[ -n "$out" ]]; then
  echo >&2 "You used ioutil.TempDir in tests, instead use testutil.NewTempDir():"
  echo >&2 $"out"
  exit 1
fi
```

wdyt?

Reply:
I would have assumed that people search for code examples how it's done. But adding a test is much better!